src/Infiltrator.jl

```diff
@@ -111,19 +111,49 @@ end
 Wraps expression in a try block, infiltrate if an exception is raised.
 Equivalent to:
 
-    try
-        expr
-    catch
-        @infiltrate
-    end
+```julia
+try
+    expr
+catch
+    @infiltrate
+end
+```
+
+If `expr` is of the form `x = f(...)`, we take `x` out of the scope 
+
+```julia
+x = try
+    f(...)
+catch
+    @infiltrate
+end  
+```
+
 """
 macro infiltry(expr)
-    return quote
-        try
-            $(esc(expr))
+    infiltrator_catch_and_rethrow = quote
+        $(Infiltrator.start_prompt)($(__module__), Base.@locals, $(String(__source__.file)), $(__source__.line), ex, catch_backtrace())
+        rethrow(ex)
```

Comment:
```suggestion
        rethrow()
```
is more correct I think.